### DISCUSSION TITLE
[Feature Request] Secondary/middle click on 'close,' 'hide,' and 'full screen' window buttons to trigger user-defined actions

### DISCUSSION THREAD

**Discussion by binford-development** [rxhanson/RectanglePro-Community#482] [Ideas]
BetterSnapTool has a feature where you can select an action and bind it to one of the three window buttons. These actions are triggered by either a right or middle click on each respective button. I've provided a couple pictures to show the feature I am explaining. While I do enjoy Rectangle Pro's rich feature set, I still find myself missing this one function. Perhaps other users may find it useful as well. 

<img width="400" alt="BetterSnap_1" src="https://github.com/rxhanson/RectanglePro-Community/assets/162678184/ef837311-2b6d-4337-be8d-e07d62aa925a">

<img width="400" alt="BetterSnap_2" src="https://github.com/rxhanson/RectanglePro-Community/assets/162678184/bf926a52-be46-42ef-ae3b-dc59993e9924">

**Comment by rxhanson** [comment]
This was requested a long time ago, and I'll be honest - I kinda forgot about it. This seems weird to include a lot of those options. What do you actually configure here? (I'm not opposed to the idea, but only some of those options make sense to me).

**Reply by binford-development** [reply]
For me, I like having the 'quit app' function mapped to the red button. Having to right click the app from the dock and select 'quit' is both annoying and far from intuitive (in my opinion). 

 I also find the 'restore old size and position' function rather useful when mapped to the yellow button as it remedies two different problems I sometimes run into: 
- some application windows have crowded toolbars - navigating to the button is quicker than looking for a space to drag the window     from
-  sometimes the window does not return to its previous size/position after un-snapping - the button typically always forces the window back to its original size/position

Lastly, I will map the drop down menu to the green button. Similar to the previous issue regarding the crowded toolbar, sometimes it is quicker to press the green button and select where I want the window to snap rather than manually dragging it there. I am using a MacBook, so there are also times when a window has to be dragged further than the trackpad allows me to drag (even with a high sensitivity).

**Reply by rxhanson** [reply]
Thanks! This makes more sense to me now. I'll add this one to my backlog and post back here when I can get to it. (Unfortunately, I don't have a timeframe for it at this time).
